ion/reader.go

```diff
@@ -98,6 +98,9 @@ type Reader interface {
 	// It returns nil if there is no current value or the current value has no annotations.
 	Annotations() []string
 
+	// AnnotationsAsSymbols returns the set of annotations associated with the current value as SymbolTokens.
+	AnnotationsAsSymbols() []SymbolToken
```

Comment:
Where does validation for symbol IDs being out of bounds take place?

If a `Reader` encounters this value when the symbol table's MaxId is `100`:
```js
$33::$0::$705::"foo"
```
what will be returned by a call to `AnnotationsAsSymbols`? Does the user get three `SymbolToken` instances and it's up to them to detect that the third one is illegal? I think I'd expect a `([]SymbolToken, error)` signature to handle that case, but want to make sure I understand what you had in mind.

Reply:
So `r.AnnotationsAsSymbols()` simply returns the reader's internal `r.Annotations` slice of type `[]SymbolToken`.

The symbol ID validation happens in the `SymbolToken` constructor during `r.Next()`. As a result the invalid `SymbolToken` never gets appended to the reader's `r.annotations` slice.

Reply:
So the error would be surfaced when the user called `r.Next()` and the `r.annotations` slice is being populated, preventing `r.AnnotationsAsSymbols()` from ever being called?